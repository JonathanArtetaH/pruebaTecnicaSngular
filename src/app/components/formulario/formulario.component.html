<div class="container">
    <input type="number" name="" id="" placeholder="Ingresar numero" min="0" max="1000" class="w-50"
        [(ngModel)]="numero"
        (ngModelChange)="emitirCambioValor()">
</div>