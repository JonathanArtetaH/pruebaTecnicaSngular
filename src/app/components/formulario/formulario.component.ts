import { Component, EventEmitter, OnInit, Output } from '@angular/core'; 
@Component({
  selector: 'app-formulario',
  templateUrl: './formulario.component.html',
  styles: [''],
})
export class FormularioComponent { 
 
  numero: number = 0;
  @Output() nuevoNumeroEvent = new EventEmitter<number>();
 
  emitirCambioValor() { 
    this.nuevoNumeroEvent.emit(this.numero);
  }

}
