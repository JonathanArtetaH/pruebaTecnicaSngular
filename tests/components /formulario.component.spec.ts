import { ComponentFixture, TestBed } from '@angular/core/testing';
import { RouterTestingModule } from '@angular/router/testing';    
import { FormularioComponent } from '../../src/app/components/formulario/formulario.component';

describe('FormularioComponent', () => {
  
  let component: FormularioComponent;
  let fixture: ComponentFixture<FormularioComponent>;
  let compiled: HTMLElement;

  beforeEach(async () => {
    await TestBed.configureTestingModule({
      imports: [
        RouterTestingModule
      ],
      declarations: [
        FormularioComponent
      ],
    }).compileComponents();
  });

  beforeEach(() => {
    fixture = TestBed.createComponent(FormularioComponent);
    component = fixture.componentInstance;
    fixture.detectChanges();
    compiled = fixture.nativeElement;
  });

  test('Creacion de componente correctamente', () => {
    expect(component).toBeTruthy();
  });

  test('Debe coincidir con el snapshot', () => {
    const fixture = TestBed.createComponent(FormularioComponent);
    fixture.detectChanges();
    const compiled = fixture.nativeElement as HTMLElement;
    expect( compiled ).toMatchSnapshot();
  })

   
});
