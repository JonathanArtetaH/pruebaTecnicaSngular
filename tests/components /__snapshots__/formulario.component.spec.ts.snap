// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormularioComponent Debe coincidir con el snapshot 1`] = `
<div
  id="root2"
>
  <div
    class="container"
  >
    <input
      class="w-50"
      max="1000"
      min="0"
      name=""
      placeholder="Ingresar numero"
      type="number"
    />
    
  </div>
</div>
`;
